catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt

```diff
@@ -0,0 +1,187 @@
+        googleTracker.send(ScreenViewBuilder().build())
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.home -> onBackPressed()
+            else -> return super.onOptionsItemSelected(item)
+        }
+        return true
+    }
+
+    private fun checkIfCrashRecoveryAndFinishActivity(activity: Activity) {
+        if (isRecoveringFromCrash) {
+            if (activity is MainMenuActivity) {
+                PreferenceManager.getDefaultSharedPreferences(this).edit()
+                    .putBoolean(RECOVERED_FROM_CRASH, false)
+                    .apply()
+            } else {
+                activity.finish()
+            }
+        }
+    }
+
+    private val isRecoveringFromCrash: Boolean
+        get() = PreferenceManager.getDefaultSharedPreferences(this)
+            .getBoolean(RECOVERED_FROM_CRASH, false)
+
+    override fun addToRequiresPermissionTaskList(task: RequiresPermissionTask) {
+        permissionRequestActivityExtension.addToRequiresPermissionTaskList(task)
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        permissionRequestActivityExtension.onRequestPermissionsResult(
+            this,
+            requestCode,
+            permissions,
+            grantResults
+        )
+    }
+
+    override fun onPause() {
+        super.onPause()
+        val pm = getSystemService(POWER_SERVICE) as PowerManager
+        if (surveyCampaign != null && (isApplicationSentToBackground(this) || !pm.isInteractive)) {
+            surveyCampaign.endAppTime(this)
+        }
+    }
+
+    private fun isApplicationSentToBackground(context: Context): Boolean {
+        val activityManager = context.getSystemService(ACTIVITY_SERVICE) as ActivityManager
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val tasks = activityManager.getRunningTasks(1)
+            val topActivity = tasks[0].topActivity
+            if (topActivity?.packageName == context.packageName) {
+                return false
+            }
+        } else {
+            val runningProcesses = activityManager.runningAppProcesses
+            for (processInfo in runningProcesses) {
+                if (processInfo.importance == RunningAppProcessInfo.IMPORTANCE_FOREGROUND) {
+                    return checkActiveProcess(processInfo, context)
+                }
+            }
+        }
+        return true
+    }
+
+    private fun checkActiveProcess(processInfo: RunningAppProcessInfo, context: Context): Boolean {
+        for (activeProcess in processInfo.pkgList) {
+            if (activeProcess == context.packageName) {
+                return false
+            }
+        }
+        return true
+    }
+
+    companion object {
+        private var savedInstanceStateExpected = false
+        const val RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH"
```

Comment:
Use (if possible private) compile-time constant instead, please.
Please check the other constants too.

Reply:
In this case private is not possible as this const used by other file ( `BaseExceptionHandler` ), I tried internal modifier please have a look :)